cardano-crypto-tests/src/Test/Crypto/VRF.hs

```diff
@@ -25,10 +26,13 @@ import Test.QuickCheck
          ((==>), (===), Arbitrary(..), Gen, Property,  NonNegative(..),
           counterexample)
 import Test.Tasty (TestTree, testGroup)
-import Test.Tasty.QuickCheck (testProperty)
+import Test.Tasty.QuickCheck (testProperty, vectorOf)
 
 {- HLINT IGNORE "Use <$>" -}
 
+-- | Newtype to generate random ByteStrings of size 32
+newtype BS32 = BS32 { unBS32 :: BS.ByteString } deriving (Show)
+
```

Comment:
Instead of newtyping `ByteString`, you can also newtype `Seed`, and use the existing `arbitrarySeedOfSize` function to define it. You could even generalize this with a type-level Nat, like so:

```haskell
newtype SizedSeed (n :: Nat) = SizedSeed { unSizedSeed :: Seed } deriving Show

instance (KnownNat n) => Arbitrary (SizedSeed n) where
    arbitrary = SizedSeed <$> arbitrarySeedOfSize (fromIntegral $ natVal (Proxy @n))
```

And then instead of having `BS32` arguments and converting them to seeds when needed, you would use `SizedSeed 32` and just `unSizedSeed` (or `coerce`, because they're newtypes) them.

For this PR, this doesn't really matter much, but in the interest of avoiding code duplication, I think having the `SizedSeed` generalization in Test.Crypto.Util would be good.

Reply:
I've added the SizedSeed at type level as you suggested.